File: 3616-make-array-elements-equal-to-zero/make-array-elements-equal-to-zero.py
class Solution:
    def countValidSelections(self, nums: List[int]) -> int:
        n = len(nums)
        left, right = 0, sum(nums)
        count = 0

        for i in range(n):
            left += nums[i]
            right -= nums[i]

            if nums[i] != 0:
                continue
            if left == right:
                count += 2
            if abs(left - right) == 1:
                count += 1

        return count

File: 3616-make-array-elements-equal-to-zero/make-array-elements-equal-to-zero.cs
public class Solution {
    public int CountValidSelections(int[] nums) {
        int n = nums.Length, count = 0, left = 0, right = nums.Sum();

        for (int i = 0; i < n; i++) {
            left += nums[i];
            right -= nums[i];

            if (nums[i] != 0) {
                continue;
            }
            if (left == right) {
                count += 2;
            }
            if (Math.Abs(left - right) == 1) {
                count++;
            }
        }

        return count;
    }
}